src/components/BiweeklySubscription.tsx

```diff
@@ -0,0 +1,110 @@
+import { FaEnvelope } from 'react-icons/fa6';
+import { emailRegex } from './Support/form/utils';
+import { Button } from './Button';
+import { useState } from 'react';
+import { cn } from '@utils/cn';
+import Loading from './Loading';
+
+const BiweeklySubscription = () => {
+  const [isValidEmail, setIsValidEmail] = useState(true);
+  const [isSubscribing, setIsSubscribing] = useState(false);
+  const [email, setEmail] = useState('');
+
+  const { apiKey, url } = (() => {
+    const apiKey = import.meta.env.PUBLIC_BEHIIV_API_KEY;
+    const url = import.meta.env.PUBLIC_BEHIIV_SUBSCRIBE_URL;
+
+    if (!apiKey || !url) {
+      throw Error(
+        `👹 Oops! Missing environment variables (url ${url}, apiKey ${apiKey})`,
+      );
+    }
+
+    return {
+      apiKey,
+      url,
+    };
+  })();
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+    if (e.target.value === '') {
+      setIsValidEmail(true);
+      return;
+    }
+
+    setIsValidEmail(emailRegex.test(e.target.value));
+  };
+
+  const handleSubscription = async () => {
+    setIsSubscribing(true);
+    try {
+      if (url && apiKey) {
+        const response = await fetch(url, {
+          method: 'POST',
+          headers: {
+            Authorization: `Bearer ${apiKey}`,
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            email,
+          }),
+        });
+
+        console.log(response);
+        return;
+      }
+
+      throw new Error('Missing environment variables');
+    } catch (error) {
+      console.log('Request failed', error);
+    } finally {
+      setIsSubscribing(false);
+    }
+  };
```

Comment:
Removed redundancy given url and apiKey throws earlier. Also, removed a console.log.

```suggestion
const handleSubscription = async () => {
  setIsSubscribing(true);
  try {
    const response = await fetch(url, {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${apiKey}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email,
      }),
    });
  } catch (error) {
    console.log('Request failed', error);
  } finally {
    setIsSubscribing(false);
  }
};

```